multiapi-engine/src/main/java/com/sngular/api/generator/plugin/openapi/OpenApiGenerator.java

```diff
@@ -224,18 +223,23 @@ private String processModelPackage(final String modelPackage) {
     return modelReturnPackage;
   }
 
-  private String processPath(final String fileSpecPackage, final Boolean isModel) {
+  private String processPath(final String fileSpecPackage, final Boolean isModel) throws IOException {
     Path path;
     final File[] pathList = Objects.requireNonNull(baseDir.listFiles(targetFileFilter));
     if (pathList.length > 0) {
       path = pathList[0].toPath().resolve(convertPackageToTargetPath(fileSpecPackage, isModel));
     } else {
       path = targetFolder.toPath();
-      path.toFile().mkdirs();
-      path = path.resolve(convertPackageToTargetPath(fileSpecPackage, isModel));
+      if (path.toFile().mkdirs()) {
+        path = path.resolve(convertPackageToTargetPath(fileSpecPackage, isModel));
+      } else {
+        throw new IOException("Problem creating folders: " + path.toFile());
+      }
     }
     if (!path.toFile().isDirectory()) {
-      path.toFile().mkdirs();
+      if (!path.toFile().mkdirs()) {
```

Comment:
First I was thinking in unify the condition `(!path.toFile().isDirectory() && !path.toFile().mkdirs() )`, but in the lines 235-237 we have the same code (and we can create a function).
```
function launchExceptionWhenNeeded(final File[] pathList)  throws IOException{
if (!path.toFile().mkdirs()) {
       throw new IOException("Problem creating folders: " + path.toFile());
      }
}
```
In a team, they prefer to wait until the third repetition to unify the code, so due as you wish.